### PR TITLE
5 - libraries:iio: Move server_ops inside iio

### PR DESCRIPTION
**Pull request by mchindri**
See last commit only
Built on top of: #664

### CODE REVIEW

File: libraries/iio/iio.c [hunk 1]
```diff
@@ -833,18 +858,31 @@ ssize_t iio_init(struct iio_desc **desc, struct iio_server_ops *iio_server_ops)
...
 
-	ops->read = iio_server_ops->read;
-	ops->write = iio_server_ops->write;
+	ldesc->phy_type = init_param->phy_type;
+	if (init_param->phy_type == USE_UART) {
+		ret = uart_init((struct uart_desc **)&ldesc->phy_desc,
+					init_param->phy_init_param);
+		if (IS_ERR_VALUE(ret))
+			goto free_desc;
+		ops->read = iio_phy_read;
```
Comment (cristiantpop) [inline libraries/iio/iio.c:867]: You can move this outside if, since we already test `init_param->phy_type` inside `iio_phy_read`

**Comment by mchindri** [comment]
v2:
 - Move server_ops attribution outside if

File: libraries/iio/iio.c (hunk 1 above)
Comment (mchindri) [reply libraries/iio/iio.c]: done

**Comment by mchindri** [comment]
v3:
 - Rebase on top of https://github.com/analogdevicesinc/no-OS/pull/664 only last commit is relevant

**Comment by mchindri** [comment]
v4: 
 - Rebase again, because  #664 was also rebased

**Comment by mchindri** [comment]
v5:
 - rebase onto master
 - fix astyle

**Comment by mchindri** [comment]
v6: Fix memory leak

File: libraries/iio/iio.c [hunk 2]
```diff
@@ -83,17 +84,39 @@ struct element_info {
 struct iio_desc {
 	struct tinyiiod			*iiod;
 	struct tinyiiod_ops		*iiod_ops;
+	enum pysical_link_type	phy_type;
```
Comment (buha) [inline libraries/iio/iio.c:87]: inconsistent indentation

File: libraries/iio/iio.c [hunk 3]
```diff
@@ -83,17 +84,39 @@ struct element_info {
...
+	void					*phy_desc;
 };
 
 /**
  * iio_read_attr(), iio_write_attr() functions, they need to know about iio_interfaces
  */
-static struct iio_interfaces *iio_interfaces = NULL;
+static struct iio_interfaces	*iio_interfaces = NULL;
+static struct iio_desc			*g_desc;
```
Comment (buha) [inline libraries/iio/iio.c:95]: inconsistent indentation

File: libraries/iio/iio.c [hunk 4]
```diff
@@ -806,12 +829,17 @@ ssize_t iio_unregister(struct iio_desc *desc,
...
+ssize_t iio_init(struct iio_desc **desc, struct iio_init_param *init_param)
 {
-	struct iio_desc *ldesc;
-	struct tinyiiod_ops *ops = (struct tinyiiod_ops *)calloc(1,
-				   sizeof(struct tinyiiod_ops));
+	int32_t			ret;
+	struct iio_desc		*ldesc;
+	struct tinyiiod_ops	*ops;
 
+	if (!init_param || init_param->phy_type == USE_NETWORK)
```
Comment (buha) [inline libraries/iio/iio.c:838]: why iio_init would fail when you use network physical layer ?

Comment (mchindri) [reply libraries/iio/iio.c:838]: Because it is not implemented yet

Comment (buha) [reply libraries/iio/iio.c:838]: Here you're supposed to check for init_param.

init_param->phy_type is checked at this line and you should add an else if (init_param->phy_type == USE_NETWORK) at this line:
https://github.com/analogdevicesinc/no-OS/blob/87a1ac9e1eea2f501b00ecb9a10c43340154c4f2/libraries/iio/iio.c#L874

**Comment by mchindri** [comment]
v7:
 - Fix indentation
 - Remove usless check
 - Remove unused struct iio_server_ops from iio_types.h

File: libraries/iio/iio.c (hunk 2 above)
Comment (mchindri) [reply libraries/iio/iio.c:87]: done

File: libraries/iio/iio.c (hunk 3 above)
Comment (mchindri) [reply libraries/iio/iio.c]: done

File: libraries/iio/iio.c (hunk 4 above)
Comment (mchindri) [reply libraries/iio/iio.c]: true, I removed the first check

File: libraries/iio/iio.c (hunk 2 above)
Comment (mirceacaprioru) [reply libraries/iio/iio.c:87]: still looks inconsistent. maybe use just tabs or just spaces.

File: libraries/iio/iio.c [hunk 5]
```diff
@@ -49,6 +49,7 @@
 #include "util.h"
 #include "error.h"
 #include "errno.h"
+#include "uart.h"
```
Comment (mirceacaprioru) [inline libraries/iio/iio.c:52]: typo in commit message: "intermadiat"

File: libraries/iio/iio.c (hunk 2 above)
Comment (mchindri) [reply libraries/iio/iio.c:87]: image

In the PR seems to be the old line.

**Comment by mchindri** [comment]
v8:
 - Fix commit type

File: libraries/iio/iio.c (hunk 5 above)
Comment (mchindri) [reply libraries/iio/iio.c:52]: done

File: libraries/iio/iio.c (hunk 2 above)
Comment (mchindri) [reply libraries/iio/iio.c:87]: But if you look at the file changes it is ok

**Pull request merged**